From: Rob Evers <revers@redhat.com>
Date: Mon, 4 Jan 2010 22:52:07 -0500
Subject: [scsi] lpfc: Fix adapter reset and off/online stress test
Message-id: <20100104225213.24386.34081.sendpatchset@localhost.localdomain>
Patchwork-id: 22300
O-Subject: [RHEL5.5 PATCH 8/18] Fixed adapter reset and offline/online stress
	test failing with I/O errors (CR: 92904)
Bugzilla: 549763
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Mike Christie <mchristi@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=549763

Fixed adapter reset and offline/online stress test failing with I/O errors (CR: 92904)

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_hbadisc.c b/drivers/scsi/lpfc/lpfc_hbadisc.c
index 223a7f4..c740fed 100644
--- a/drivers/scsi/lpfc/lpfc_hbadisc.c
+++ b/drivers/scsi/lpfc/lpfc_hbadisc.c
@@ -873,6 +873,10 @@ lpfc_linkdown(struct lpfc_hba *phba)
 
 	if (phba->link_state == LPFC_LINK_DOWN)
 		return 0;
+
+	/* Block all SCSI stack I/Os */
+	lpfc_scsi_dev_block(phba);
+
 	spin_lock_irq(&phba->hbalock);
 	phba->fcf.fcf_flag &= ~(FCF_AVAILABLE | FCF_DISCOVERED);
 	if (phba->link_state > LPFC_LINK_DOWN) {
